### PR TITLE
you-have-mail-cli: init at 0.3.0

### PR DESCRIPTION
**Pull request by Golbinex**
## Description of changes
[You Have Mail](https://github.com/LeanderBB/you-have-mail-cli) is an application designed for notifying the user of new emails via StdOut or ntfy notifications. This is useful for receiving notifications from mailboxes which do not support UnifiedPush notifications nor unofficial email clients, for example ProtonMail.

## Things done
Prepared package.nix for successful assembly and installation of the app.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Golbinex** [comment]
Result of nixpkgs-review run on x86_64-linux

1 package built:
you-have-mail-cli

File: pkgs/by-name/yo/you-have-mail-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    hash = "sha256-YgIztCmL+uJUUXu+BWXLf9f4L7UpuPf0onsfMRtlWkw=";
+  };
+
+  nativeBuildInputs = [
+    libclang.lib
+    llvmPackages.libcxxClang
+    go
+  ];
+
+  LIBCLANG_PATH = "${libclang.lib}/lib";
```
Comment (Aleksanaa) [inline pkgs/by-name/yo/you-have-mail-cli/package.nix:28]: You may want to add `rustPlatform.bindgenHook` to `nativeBuildInputs` and remove all things about clang.

File: pkgs/by-name/yo/you-have-mail-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    outputHashes = {
+      "you-have-mail-common-0.8.1" = "sha256-jDOIMvHqWvPpnyZ5V+6VsyFUDXz84yFROkInXE5CIyE=";
+    };
+  };
+
+  meta = {
+    description = "Small application to notify you when you receive an email in your email account";
+    homepage = "https://github.com/LeanderBB/you-have-mail-cli";
+    license = lib.licenses.agpl3Only;
+    maintainers = with lib.maintainers; [ baksa ];
```
Comment (Aleksanaa) [inline pkgs/by-name/yo/you-have-mail-cli/package.nix:63]: missing `mainProgram`

File: pkgs/by-name/yo/you-have-mail-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  llvmPackages,
+  libclang,
+  go,
+  buildGoModule,
+}:
+
+rustPlatform.buildRustPackage rec {
```
Comment (Aleksanaa) [inline pkgs/by-name/yo/you-have-mail-cli/package.nix:11]:
```suggestion
rustPlatform.buildRustPackage {
```

Not used?

**Comment by Golbinex** [comment]
Thanks, suggested changes applied.

**Review by Aleksanaa — APPROVED** [review]
Waiting for CI

**Pull request merged**